Big Paperwork: Adds the Paperwork loan offer, paperwork fraud, and ancient paperwork!

Rhials opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Adds paperwork -- an indecipherable mess of papers that only a seasoned department head could hope to decipher. If processed and sent to Central Command via the cargo shuttle, the budget gets a bonus.

Paperwork is distributed through a new Shuttle Loan offer. A nearby station needs some paperwork reviewed. Do you have what it takes to handle BIG PAPERWORK?

Processing paperwork is relatively simple. You can inspect it to see which head of staff it "belongs" to, and you can use their stamp to complete it. You get a lot of these, so the bonuses can add up if you get them all in. Just be careful not to return any unprocessed paperwork.

With big paperwork comes big paperwork fraud, and big fines.

![image](https://user-images.githubusercontent.com/28870487/198452450-86712aff-440e-4a66-88ea-69094f401986.png)

You can photocopy your completed paperwork to double-dip with Central Command and squeeze out a little bit more money. The more paperwork fraud you commit, the higher the chance you'll be caught and fined.

On top of that:

![tboi unlock](https://user-images.githubusercontent.com/28870487/198710944-7479078b-0426-4f7f-9360-40d3b0b03c74.png)

Some say that in the long forgotten halls of maintenance, there lies _ancient paperwork_. Documents whose importance has long since passed, but are still important for bookkeeping at Central Command. With a keen eye, you might spot one of these while scouring through maintenance. Make sure it gets forwarded to the right person, cargo will thank you.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

More shuttle loans offers means more event variation.

Provides convenient access to the bottom layer of hell (Fraud) for cargo staff. 

Adds some rare maintenance loot that, with a tiny bit of effort, benefits the crew. It's like a side quest for assistants.

<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
add: Paperwork! Huge paperwork. Gigantic. You can stamp it and send it to Centcom via cargo for BIG MONEY.
add: New shuttle loan event variant -- You get a bunch of paperwork!
add: Ancient paperwork has appeared in the maintenance tunnels!
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Rhials committed:
papers please

Rhials committed:
the event is no longer a one-time deal

the max_occurences value hasn't been touched in 9 years. I don't see a reason why this one shouldn't be able to run more than once per round. It presents cargo with a risk/reward choice, and can introduce many unique and interesting elements to a round

Rhials committed:
oops I forgot to include the actual change I was talking about in the last push

Rhials committed:
big pr wow

paperwork now exists

it can be stamped, subtypes work

Rhials committed:
new name

it works wow

Rhials committed:
ok cool

Rhials committed:
oh WOW theres an entire folder for this??

Rhials committed:
stamps are implemented but dont work

stamping works, desc works, lotta progress

Rhials committed:
big text wow

Rhials committed:
update to the papers

Rhials committed:
overlays really work this time!

Rhials committed:
better messages

Rhials committed:
better examine functionality, better feedback, papa johns

also, random paperwork exists now wow!

Rhials committed:
slight changes to how stuff is spawned

Rhials committed:
adds copier functionality and copy subtypes

Rhials committed:
name change

Rhials committed:
renames the paperwork file names again.

also undoes some debug code in an attempt to figure out what I'm doing wrong here

Rhials committed:
fixes some coding missteps, changes how bonus text works

the extra examine text is now loaded onto the object at initialize, and tacks it onto the end of the examine text in examine_more

Rhials committed:
photocopy progress

changes how paperwork is selected (back to 1 for everyone!!)

photocopies work now I hope

adds a grey stamp for copy/pasting

Rhials committed:
further text updates and photocopy fixes

Rhials committed:
(some) fixes to photocoping and yet ANOTHER rework of examine functionality

photocopying works now but photocopies suck and dont transfer the stamp

Rhials committed:
add_stamp is now used to handle stamping

photocopier now (should) copy stamps,
better voiding logic

Rhials committed:
modifies some text to better fit the joke

Rhials committed:
minor text changes, paperwork now fits in folders

Rhials committed:
redoooooo

Rhials committed:
stamp_overlay now generates on add_stamp

also grabs the initials for ancient paperwork wow

Rhials committed:
final (hopefully) grammatical touch-ups on paperwork messages

Rhials committed:
reworks how photocopies and ancient paperwork work

they now use copy_stamp_info to grab some vars and use them as their own, effectively making any photocopy/ancient subtype able to "mutate" into a less detailed version of a paperwork piece

Rhials committed:
no more inspection text on ancient papers

detailed_desc now establishes itself in initialize completely, to prevent subtypes from having the "as you sift through etc etc." text

Rhials committed:
formatting consistency

Rhials committed:
finishes the exports page, adds photocopy export

rewrites medical again too

Rhials committed:
backfiring is now reported on and stops when a backfire hits

Rhials committed:
event now actually spawns the paperwork,

Rhials committed:
testing fixes

ancient paperwork no longer copies photcopy/ancient papers

event paper spawning logic now uses istype (now the other paper types dont spawn)

Rhials committed:
better cost calculation logic

Rhials committed:
ancient paperwork now spawns in maint and doesn't pick the wrong subtypes to copy

Rhials committed:
FINALLY fixes the photocopy bug

examine_more for photocopies now uses spans

Rhials committed:
more spans, better grammar. Removes canspawnevent

canspawnevent change should go in another PR because its a balance thing I guess

Rhials committed:
tweaks some values, photocopies cannot be photocopied

arghghg

Rhials committed:
removes some debug come

imagine pushing a commit you know is going to cause a merge conflict the second you post the pr

Rhials committed:
fixes autodoc comment

Rhials committed:
adds a comment, clears some lines

Rhials committed:
comment

Rhials committed:
oh my god fine

Rhials committed:
better descs

Rhials committed:
comment

Rhials committed:
removes extra comment

private-tristan commented:
The Binding of Spessmen

Rhials committed:
oops

GuiltyNeko commented:
I love this it's absolutely fantastic, please merg.
Also I noticed that these all have the same first detailed description line, is it possible to do code magic to just write that once and have it apply to all of them to reduce the line count? Or would that be too "point A and point B too far separated from each other and now confusing"?

Rhials committed:
initial touchups

Jackal-boop commented:
Holy shit rhials

Rhials committed:
????

Rhials committed:
remove , fixes more stuff

Rhials committed:
smite thursdays!!

Rhials committed:
oops

bruhman8ghughug commented:
i love when stuff in the lore (tax fraud) gets reflected on gameplay

Rhials committed:
more fixes

overlay changes

Rhials committed:
this too

I want to get this in and have all of the typecache business in a seperate commit in case I break anything

Rhials committed:
it works wow

Rhials committed:
why yes, yes it is

Rhials committed:
wow!

Rhials committed:
form atting

Rhials committed:
give me a fucking break mister linter

Rhials committed:
wow

Rhials committed:
Merge remote-tracking branch 'upstream/master' into new-shuttle-thing-

Rhials committed:
why was that there?

Rhials committed:
suicide act wow

Twaticus reviewed (changes requested):
![chrome_Yr860DJIig](https://user-images.githubusercontent.com/46540570/201823515-1708c028-2d4e-4509-b398-2e36454e9884.png)

Rhials committed:
Merge remote-tracking branch 'upstream/master' into new-shuttle-thing-

Rhials committed:
proc ref fix wow

Rhials committed:
fix??

MrMelbert reviewed (changes requested):
Very small things but looks good to merge after

Mothblocks commented on code/modules/events/shuttle_loan.dm:
```diff
@@ -277,6 +287,11 @@
 					shuttle_spawns.Add(/obj/item/paper/fluff/cargo/bomb/allyourbase)
 
+			if(PAPERS_PLEASE)
+				for(var/new_paperwork in subtypesof(/obj/item/paperwork))
+					if(!ispath(new_paperwork, /obj/item/paperwork/photocopy) && !ispath(new_paperwork, /obj/item/paperwork/ancient))
+						shuttle_spawns.Add(new_paperwork)
```
```suggestion
				shuttle_spawns += subtypesof(/obj/item/paperwork) - typesof(/obj/item/paperwork/photocopy) - typesof(/obj/item/paperwork/ancient)
```

GuiltyNeko commented on code/modules/paperwork/paperwork.dm:
```diff
@@ -0,0 +1,252 @@
+/obj/item/paperwork/service/Initialize()
+	. = ..()
+
+	detailed_desc += span_info(" Your begin scanning over the document. This is a standard Nanotrasen NT-435Z3 form used for requests to Central Command.")
```
Your Begin

Rhials committed:
back and ready to code

Rhials committed:
new suicide, moves the overlay stuff to add_overlay

for some reason update overlays really did not want to work the way I wanted it to so im just oding it like this

Rhials committed:
misc fixes

Rhials committed:
minor attackby changes, cham stamp stuff

Rhials committed:
oops

DAKKA-WAAAGH commented:
YEAAAAHHH!!
